# Dépendance

In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib as mt
import matplotlib.pyplot as plt

# Controle 

1- C'est une clé permettant d'indiquer le progrès réalisé ou de mesurer le progrès à réaliser pour atteindre un objectif stratégique. Par exemple le taux de visite d'un site web est un KPI.

In [2]:
data = pd.read_csv('./data/eval_data.csv')

In [15]:
data.head(10)

Unnamed: 0  order_id  quantity                              item_name  \
0           0         1         1           Chips and Fresh Tomato Salsa   
1           1         1         1                                   Izze   
2           2         1         1                       Nantucket Nectar   
3           3         1         1  Chips and Tomatillo-Green Chili Salsa   
4           4         2         2                           Chicken Bowl   
5           5         3         1                           Chicken Bowl   
6           6         3         1                          Side of Chips   
7           7         4         1                          Steak Burrito   
8           8         4         1                       Steak Soft Tacos   
9           9         5         1                          Steak Burrito   

                                  choice_description item_price  
0                                                NaN        NaN  
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN        NaN  
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
5  [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98   
6                                                NaN     $1.69   
7  [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75   
8  [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...     $9.25   
9  [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...     $9.25

In [4]:
data.ndim
#Dimension de la dataset

2

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 6 columns):
Unnamed: 0            4622 non-null int64
order_id              4622 non-null int64
quantity              4622 non-null int64
item_name             4622 non-null object
choice_description    3376 non-null object
item_price            4522 non-null object
dtypes: int64(3), object(3)
memory usage: 216.7+ KB


In [9]:
most = data.groupby(['item_name'])['quantity'].count()

In [32]:
most

item_name
6 Pack Soft Drink                         54
Barbacoa Bowl                             66
Barbacoa Burrito                          91
Barbacoa Crispy Tacos                     11
Barbacoa Salad Bowl                       10
Barbacoa Soft Tacos                       25
Bottled Water                            162
Bowl                                       2
Burrito                                    6
Canned Soda                              104
Canned Soft Drink                        301
Carnitas Bowl                             68
Carnitas Burrito                          59
Carnitas Crispy Tacos                      7
Carnitas Salad                             1
Carnitas Salad Bowl                        6
Carnitas Soft Tacos                       40
Chicken Bowl                             726
Chicken Burrito                          553
Chicken Crispy Tacos                      47
Chicken Salad                              9
Chicken Salad Bowl                       110


8- Nous pouvons voir que l'item le plus vendu est le Chicken Bowl avec 726 entrée. 

In [13]:
data['quantity'].count()

4622

9 - 4622 On était commandé au total.

In [26]:
test = 0
for x in data : 
    test += x['quantity'] * x['item_price']

TypeError: string indices must be integers, not str